src/soc/ibm/power9/istep_14_3.c

```diff
@@ -277,26 +281,26 @@ static void init_phbs(uint8_t phb_active_mask, const uint8_t *iovalid_enable)
 	};
 
 	/* ATTR_PROC_PCIE_MMIO_BAR0_BASE_ADDR_OFFSET */
-	uint64_t mmio_bar0_offsets[MAX_PHB_PER_PROC] = { 0 };
+	const uint64_t mmio_bar0_offsets[MAX_PHB_PER_PROC] = { 0 };
 	/* ATTR_PROC_PCIE_MMIO_BAR1_BASE_ADDR_OFFSET */
-	uint64_t mmio_bar1_offsets[MAX_PHB_PER_PROC] = { 0 };
+	const uint64_t mmio_bar1_offsets[MAX_PHB_PER_PROC] = { 0 };
 	/* ATTR_PROC_PCIE_REGISTER_BAR_BASE_ADDR_OFFSET */
-	uint64_t register_bar_offsets[MAX_PHB_PER_PROC] = { 0 };
+	const uint64_t register_bar_offsets[MAX_PHB_PER_PROC] = { 0 };
 	/* ATTR_PROC_PCIE_BAR_SIZE */
-	uint64_t bar_sizes[3] = { 0 };
+	const uint64_t bar_sizes[3] = { 0 };
 
 	/* Determine base address of chip MMIO range */
 	uint64_t base_addr_mmio = 0;
-	base_addr_mmio |= PPC_PLACE(0, 8, 5);  // ATTR_PROC_FABRIC_SYSTEM_ID
-	base_addr_mmio |= PPC_PLACE(0, 15, 4); // ATTR_PROC_EFF_FABRIC_GROUP_ID
-	base_addr_mmio |= PPC_PLACE(0, 19, 3); // ATTR_PROC_EFF_FABRIC_CHIP_ID
-	base_addr_mmio |= PPC_PLACE(3, 13, 2); // FABRIC_ADDR_MSEL
-	                                       // nm = 0b00/01, m = 0b10, mmio = 0b11
+	base_addr_mmio |= PPC_PLACE(0, 8, 5);     // ATTR_PROC_FABRIC_SYSTEM_ID
+	base_addr_mmio |= PPC_PLACE(chip, 15, 4); // ATTR_PROC_EFF_FABRIC_GROUP_ID
+	base_addr_mmio |= PPC_PLACE(0, 19, 3);    // ATTR_PROC_EFF_FABRIC_CHIP_ID
```

Comment:
Is it always set this way or does it depend on `MBOX_SCRATCH_REG6_GROUP_PUMP_MODE`?

In any case please add a comment mentioning that chip is not written to CHIP_ID and this is not an error (unless it is).

Reply:
Not an error. Depends on group pump mode.

I was wondering whether pump mode can vary for Nimbus at all and thus whether it makes sense to support both options? If both are potentially relevant, might as well extract a function like for DD, because the check is needed in multiple places and their number grows. If both are necessary, we'll need to verify that the check wasn't skipped anywhere else (compare against HB uses of the corresponding attribute).